cirq-rigetti/cirq_rigetti/aspen_device_test.py

```diff
@@ -204,15 +202,16 @@ def test_rigetti_qcs_aspen_device_invalid_qubit(
         device.validate_operation(cirq.I(qubit))
 
 
-def test_rigetti_qcs_aspen_device_non_existent_qubit(qcs_aspen8_isa: InstructionSetArchitecture):
+def test_rigetti_qcs_aspen_device_readonly_nodes(qcs_aspen8_isa: InstructionSetArchitecture):
     """test RigettiQCSAspenDevice throws error when qubit does not exist on device"""
     # test device may only be initialized with Aspen ISA.
     device_with_limited_nodes = RigettiQCSAspenDevice(
-        isa=InstructionSetArchitecture.from_dict(qcs_aspen8_isa.to_dict())
+        isa=InstructionSetArchitecture.from_raw(qcs_aspen8_isa.json())
     )
-    device_with_limited_nodes.isa.architecture.nodes = [Node(node_id=10)]
-    with pytest.raises(UnsupportedQubit):
-        device_with_limited_nodes.validate_qubit(cirq.GridQubit(0, 0))
+    assert len(device_with_limited_nodes.isa.architecture.nodes) > 0
+
+    device_with_limited_nodes.isa.architecture.nodes = []
+    assert len(device_with_limited_nodes.isa.architecture.nodes) > 0, 'Nodes should be read-only'
```

Comment:
If Architecture.nodes is a property, can it perhaps be defined without setter?

Reply:
Actually, this is up to https://github.com/rigetti/qcs-sdk-rust/blob/main/crates/python/src/qpu/isa.rs#L73 and I don't think it makes sense to be making assertions about these types in this library so I will remove this test.

Reply:
Removed this test - this behavior is up to `qcs_sdk`